subprojects/kotlin-dsl-integ-tests/src/integTest/kotlin/org/gradle/kotlin/dsl/integration/PrecompiledScriptPluginIntegrationTest.kt

```diff
@@ -815,4 +815,20 @@ class PrecompiledScriptPluginIntegrationTest : AbstractPluginIntegrationTest() {
 
         compileKotlin()
     }
+
+    @Test
+    fun `does not add extra task actions to kotlin compilation task`() {
+        assumeNonEmbeddedGradleExecuter()
+        withKotlinDslPlugin().appendText("""
+            gradle.taskGraph.whenReady {
+                val compileKotlinActions = allTasks.single { it.path == ":compileKotlin" }.actions.size
+                require(compileKotlinActions == 1) {
```

Comment:
Maybe use `check` instead? Probably `assert` would be best, but I guess it's not safe to assume assertions will always be enabled. `require` feels weird, and out of place.

Reply:
`require` is basically the equivalent of `assert` in Kotlin
We use often it in under-test build scripts.